projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -427,6 +427,11 @@ private void OnShutdown(ShutdownEventArgs reason)
 
         private bool SetCloseReason(ShutdownEventArgs reason)
         {
+            if (reason is null)
+            {
+                throw new ArgumentNullException(nameof(reason));
+            }
+
```

Comment:
```csharp
ArgumentNullException.ThrowIfNull(reason);
```

but this could not build on .NET Framework

Reply:
You can simplify this code by using https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-8.0

Reply:
No, we can't. This project targets `netstandard2.0` as well as `net6.0`

Reply:
This can be internally implemented. There's nothing especial in [ArgumentNullException.ThrowIfNull](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ArgumentNullException.cs,432088dc6d98b845).

But the pattern allows to inline the `if`. Any method that has a `throw` is not inlined.

[Performance Improvements in .NET 8 - Exceptions](https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#exceptions)

Reply:
If someone wants to open a PR with these sorts of changes, using `#if ... #else ... #endif` to accommodate `netstandard2.0`, by all means, do so.

Drive-by reviews on closed PRs are not going to turn into actionable work.